runner.py

```diff
@@ -0,0 +1,200 @@
+import asyncio
+from collections import namedtuple
+import logging
+import functools
+from typing import Any, Optional
+import urllib.parse
+
+import docker
+import aiohttp as ah
+from database import Database
+
+from config import settings
+
+doc = docker.from_env()
+
+logger = logging.getLogger(__name__)
+
+
+async def run_cmd(cmd: str, env: dict[str, str], vols: dict[str, dict[str, str]]) -> str:
```

Comment:
`dict[str, dict[str, str]]` is a very uninformative type, could we add a typealias to explain what it is a bit better?
```py
DescriptiveName: TypeAlias = dict[str, str]

vols: dict[str, Descriptivename]
```